x-pack/plugins/observability_solution/entity_manager/server/lib/entities/ingest_pipeline/generate_latest_processors.ts

```diff
@@ -38,16 +38,24 @@ function createMetadataPainlessScript(definition: EntityDefinition) {
 }
 
 function liftIdentityFieldsToDocumentRoot(definition: EntityDefinition) {
-  return definition.identityFields.map((identityField) => {
-    const optionalFieldPath = identityField.field.replaceAll('.', '?.');
-    const assignValue = `ctx.${identityField.field} = ctx.entity.identity.${identityField.field}.keySet().toArray()[0];`;
-    return {
-      script: {
-        if: `ctx.entity.identity.${optionalFieldPath} != null && ctx.entity.identity.${identityField.field}.size() != 0`,
-        source: cleanScript(`${initializePathScript(identityField.field)}\n${assignValue}`),
-      },
-    };
-  });
+  return definition.identityFields
+    .map((identityField) => {
+      return [
+        {
+          dot_expander: {
+            field: identityField.field,
+            path: `entity.identity.${identityField.field}`,
+          },
```

Comment:
A field like `log.logger` in the `top_metrics` end up like:
```
entity: {
  identity: {
    log: {
      logger: {
        log.logger: 'value'
      }
    }
  }
}
```

So in order to use the variable expansion of `set` we need to first expand that field.